aten/src/ATen/native/cudnn/MHA.cpp

```diff
@@ -292,6 +292,21 @@ auto fixSizeOneDimStrideSDPA(
   }
   return strides;
 }
+
+void alloc_with_matching_layout(const Tensor& q, Tensor& output, const std::vector<int64_t>& shape) {
+  TORCH_CHECK(shape.size() == q.sizes().size(), "cuDNN SDPA alloc_with_matching_layout got requested shape ndim != q ndim");
+  // get the "fill order," which is just an argsort on the strides
+  std::vector<int> fill_order(shape.size());
+  std::iota(fill_order.begin(), fill_order.end(), 0);
+  std::stable_sort(fill_order.begin(), fill_order.end(), [&q](int idx1, int idx2) {return q.strides()[idx1] < q.strides()[idx2];});
```

Comment:
nit: probably optimized away but maybe just create const auto q_strides = q.strides() and use that